speaker/main.go

```diff
@@ -110,14 +147,62 @@ func main() {
 	})
 	if err != nil {
 		logger.Log("op", "startup", "error", err, "msg", "failed to create k8s client")
+		os.Exit(1)
 	}
 	ctrl.client = client
 
-	if err := client.Run(); err != nil {
+	if mlist != nil {
+		go watchMemberListEvents(logger, eventCh, stopCh, client)
+
+		iplist, err := client.GetPodsIPs(*mlNamespace, *mlLabels)
+		if err != nil {
+			logger.Log("op", "startup", "error", err, "msg", "failed to get PodsIPs")
+			os.Exit(1)
+		}
+		n, err := mlist.Join(iplist)
+		logger.Log("op", "startup", "msg", "Memberlist join", "nb joigned", n, "error ?", err)
+		defer func() {
+			logger.Log("op", "shutdown", "msg", "leaving MemberList cluster")
+			err = mlist.Leave(time.Second)
+			logger.Log("op", "shutdown", "msg", "left MemberList cluster", "error ?", err)
+			mlist.Shutdown()
+			logger.Log("op", "shutdown", "msg", "MemberList shutdown", "error ?", err)
+		}()
+	}
+
+	if err := client.Run(stopCh); err != nil {
 		logger.Log("op", "startup", "error", err, "msg", "failed to run k8s client")
 	}
 }
 
+func watchMemberListEvents(logger gokitlog.Logger, eventCh chan memberlist.NodeEvent, stopCh chan struct{}, client *k8s.Client) {
+	for {
+		select {
+		case e := <-eventCh:
+			logger.Log("msg", "Node event", "node addr", e.Node.Addr, "node name", e.Node.Name, "node event", e.Event)
```

Comment:
Any chance we could translate the e.Event to more readable value? Currently it is just an ID of an enum.

Reply:
done